src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogGraphListener.java

```diff
@@ -173,4 +180,21 @@ long getTime(FlowNode startNode, FlowNode endNode) {
         }
         return 0;
     }
+    
+    String getResultTag(@Nonnull FlowNode endNode) {
+        ErrorAction error = endNode.getError();
+        if (error != null) {
+            return "ERROR";
+        }
+        WarningAction warningAction = endNode.getPersistentAction(WarningAction.class);
+        if (warningAction != null) {
+            Result result = warningAction.getResult();
+            return result.toString();
+        }
+        // Other possibilities are queued, launched, unknown: https://javadoc.jenkins.io/plugin/workflow-api/org/jenkinsci/plugins/workflow/actions/QueueItemAction.QueueState.html
```

Comment:
Where do these other states get returned?  Also, where do "SUCCESS" and "NOT_BUILT" get returned from?

Reply:
`QueueItemAction.getNodeState(endNode)` could return queued, launched, or unknown, but I'm only explicitly checking for cancelled because queued and launched don't reflect a real "result", so they can all be grouped into unknown

Reply:
SUCCESS and NOT_BUILT come from the above `warningAction.getResult() ` (they are all `Result`s) https://javadoc.jenkins-ci.org/hudson/model/Result.html

Reply:
Ah, okay -- `WarningAction` seems like an odd name in that case!

Would be helpful if you could add a comment to that stanza indicating what some of the expected values might be from there.

Reply:
Sure, sounds good!